tidy3d/components/lumped_element.py

```diff
@@ -281,19 +335,461 @@ def to_structure(self) -> Structure:
+        dt : float
+            Time step of the FDTD simulation.
+
+        Returns
+        -------
+        np.ndarray
+            The equivalent frequency-dependent portion of the electric permittivity.
+        """
+
+        # For fitting with a pole-residue model, we provide a convenience function for
+        # converting the complex conductivity to a complex permittivity.
+        sigma = NetworkConversions.complex_conductivity(a, b, freqs, dt)
+        return 1j * sigma / (2 * np.pi * freqs * EPSILON_0)
+
+
+class LumpedRLC(RectangularLumpedElement):
+    """Lumped element representing a circuit consisting of a resistor, capacitor, and inductor.
+
+    Notes
+    -----
+
+        Implementation is based on the equivalent medium introduced by _`[1]`.
+
+        **References**
+
+        .. [1]  J. A. Pereda, F. Alimenti, P. Mezzanotte, L. Roselli and R. Sorrentino, "A new algorithm
+                for the incorporation of arbitrary linear lumped networks into FDTD simulators," IEEE
+                Trans. Microw. Theory Tech., vol. 47, no. 6, pp. 943-949, Jun. 1999.
+    """
+
+    resistance: Optional[pydantic.PositiveFloat] = pydantic.Field(
+        None,
+        title="Resistance",
+        description="Resistance value in ohms.",
+        unit=OHM,
+    )
+
+    capacitance: Optional[pydantic.PositiveFloat] = pydantic.Field(
+        None,
+        title="Capacitance",
+        description="Capacitance value in farads.",
+        unit=FARAD,
+    )
+
+    inductance: Optional[pydantic.PositiveFloat] = pydantic.Field(
+        None,
+        title="Inductance",
+        description="Inductance value in henrys.",
+        unit=HENRY,
+    )
+
+    circuit_topology: Optional[Literal["series", "parallel"]] = pydantic.Field(
+        "series",
+        title="Circuit Topology",
+        description="Describes whether circuit elements are connected in ``series`` or ``parallel``.",
+    )
+
+    @cached_property
+    def _number_circuit_elements(self) -> pydantic.PositiveInt:
+        num_elements = 0
+        if self.resistance:
+            num_elements += 1
+        if self.capacitance:
+            num_elements += 1
+        if self.inductance:
+            num_elements += 1
+        return num_elements
+
+    @cached_property
+    def to_admittance_function(self) -> tuple[tuple[float, ...], tuple[float, ...]]:
+        """Converts the :class:`LumpedRLC` instance into a rational expression representing the
+        admittance of the network in the Laplace domain.
+        """
+        R = self.resistance
+        C = self.capacitance
+        L = self.inductance
+        if self._number_circuit_elements == 1:
+            # Lumped element is simply a resistor, inductor, or capacitor.
+            if R:
+                return ((1, 0), (R, 0))
+            if C:
+                return ((0, C), (1, 0))
+            if L:
+                return ((1, 0), (0, L))
+        elif self.circuit_topology == "series":
```

Comment:
in old days when we were still using pylint, it will complain that one should use "if" rather than "elif" here, as the function returns in the previous branching